drivers/sensor/tmp108/tmp108.c

```diff
@@ -174,7 +175,8 @@ static int tmp108_attr_get(const struct device *dev,
 	case SENSOR_ATTR_CONFIGURATION:
 		result =  tmp108_reg_read(dev,
 					  TI_TMP108_REG_CONF,
-					  (uint16_t *) &(val->val1));
+					  &tmp_val);
+		val->val1 = tmp_val;
```

Comment:
May be a good idea of adding a `val->val2 = 0;` while at it.

Reply:
Agreed.

Reply:
it makes sense